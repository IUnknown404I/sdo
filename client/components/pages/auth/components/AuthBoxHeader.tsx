import React from 'react';
import Box from '@mui/material/Box';

const AuthBoxHeader = (props: { children: JSX.Element | JSX.Element[] }) => {
	return (
		<Box
			sx={{
				position: 'relative',
				display: 'flex',
				gap: '.75rem',
				paddingBottom: '.5rem',
				flexDirection: 'row',
				alignItems: 'center',
				justifyContent: 'space-between',
				img: {
					filter: 'grayscale(.35)',
					maxWidth: '75px',
					maxHeight: '75px',
				},
				svg: {
					maxWidth: '50px',
					maxHeight: '50px',
					userSelect: 'none',
					marginRight: '-5px',
					fill: '#757575',
					transition: 'fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms',
					flexShrink: '0',
				},
			}}
		>
			{props.children}
		</Box>
	);
};

export default AuthBoxHeader;
