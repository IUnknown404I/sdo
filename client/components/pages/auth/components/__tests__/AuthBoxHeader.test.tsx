/**
 * @jest-environment jsdom
 */

import { render, waitFor } from '@testing-library/react';
import React from 'react';
import AuthBoxHeader from '../AuthBoxHeader';

const TestElement = (
	<AuthBoxHeader>
		<span id='test-span'></span>
		<span id='test-span'></span>
		<span id='test-span'></span>
	</AuthBoxHeader>
);

describe('AuthBoxHeader ->', () => {
	it('array render check:', () => {
		const { findAllByTestId } = render(TestElement);
		waitFor(() => expect(findAllByTestId('test-1')).toHaveLength(3));
	});

	it('snapshot test:', () => {
		const { container } = render(TestElement);
        expect(container).toMatchSnapshot();
	});
});
