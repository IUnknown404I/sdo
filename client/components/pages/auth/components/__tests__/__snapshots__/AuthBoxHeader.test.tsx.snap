// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthBoxHeader -> snapshot test: 1`] = `
<div>
  <div
    class="MuiBox-root css-1snj1at"
  >
    <span
      id="test-span"
    />
    <span
      id="test-span"
    />
    <span
      id="test-span"
    />
  </div>
</div>
`;
